### PR TITLE
simplify the SetupHandlers test

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Simplify the test setup for `TestSetupHandlers`

### Motivation

This test validates that dynamic routes are added vie the api providers. We previously modified the the different mock to add an api provider. That makes understanding the test harder, as is not obvious at first glance were the routes are comming from. 

By localizing the test code in the same file is easier to see. Also, reduces the amount of code for some our component mocks.

At the end the test is testing the same logic with less code.

The e2e tests are taking care of validating that each component is adding the right route. 
Example test for the `flare` endpoint:
Flare endpoint definition: https://github.com/DataDog/datadog-agent/blob/main/comp/core/flare/flare.go#L101
e2e test for flare: https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/tests/agent-subcommands/flare/flare_nix_test.go#L41-L42

### CODE REVIEW

**Review by davidor — APPROVED** [review]
:+1: for container-platform files

**Review by tobz — APPROVED** [review]

**Review by rayz — APPROVED** [review]

**Review by chouquette — APPROVED** [review]

File: comp/api/api/apiimpl/internal/agent/agent_test.go
Comment (pgimalac) [inline comp/api/api/apiimpl/internal/agent/agent_test.go:1]: Considering it's a unit test I agree it makes sense not to test the real endpoints but just a fake one, and we already have an e2e test checking the api of the agent anyway

**Review by pgimalac — APPROVED** [review]
LGTM ! Much simpler now !

File: comp/api/api/apiimpl/internal/agent/agent_test.go
Comment (GustavoCaso) [reply comp/api/api/apiimpl/internal/agent/agent_test.go:1]: We were not even testing the real endpoints 😄 We created the mock endpoint in the component's mock. So we had a bit extra complexity laying around

**Pull request merged**